Source/DafnyCore/AST/Expressions.cs

```diff
@@ -1035,7 +1035,12 @@ public DatatypeValue(IToken tok, string datatypeName, string memberName, List<Ex
   }
 
   public override IEnumerable<Expression> SubExpressions {
-    get { return Arguments; }
+    get {
+      return Bindings.Arguments != null ? Arguments :
+        Bindings.ArgumentBindings != null ? Bindings.ArgumentBindings.Select(binding => binding.Actual) :
+        new List<Expression>() { };
+      ;
```

Comment:
```suggestion
      return Arguments ??
        Bindings.ArgumentBindings?.Select(binding => binding.Actual) ??
        Enumerable.Empty<Expression>();
```

Reply:
(I think this is correct, but I didn't test it)

Reply:
It is correct. I also added a comment to justify the transformation